Update aggregation queries to use JSON columns

sethfitz opened the pull request:
* Update server to match JSONB-based schema
* Ignore `GZIPPED` (tiles should always be gzipped)
* Make page sizes consistent
* Make pagination 1-based
* Output `Timestamp`s as ISO-8601 strings

sethfitz committed:
Update aggregation queries to use JSON columns

sethfitz committed:
refreshments schema

sethfitz committed:
Generate JSONB tables

sethfitz committed:
Drop unnecessary tables

sethfitz committed:
Allow password to be omitted

sethfitz committed:
Misc: see full message

* Rename StatsRouter -> DefaultRouter
* Enable CORS
* Fix ChangesetCountry model / endpoint
* Add support for JSONB columns
* Make page sizes consistent
* Make pagination 1-based
* Make output formatting consistent (snake_case)
* Output Timestamps as ISO-8601 strings (using Instants internally)
* scalafmt

sethfitz committed:
Tiles should always be gzipped

mojodna commented:
Fixes #19

mojodna commented:
/cc @kamicut

sethfitz commented on sql/country_statistics.sql:
```diff
@@ -1,117 +1,140 @@
+  ),
+  hashtag_counts AS (
+    SELECT
+      -- TODO rank by edit count?
```
@giblet22 this produces the top 10 hashtags by number of changesets (and surfaces that count in the response). Is that the correct thing to sort by? (It could also make sense to sort by number of edits.)

giblet22 replied:
Will confirm with customer, but I agree top hashtags by edits would make sense for a better picture overall.

sethfitz replied:
👍

sethfitz commented on sql/country_statistics.sql:
```diff
@@ -1,117 +1,140 @@
+  ),
+  user_counts AS (
+    SELECT
+      -- TODO rank by edit count?
```
@giblet22 this produces the top 10 users by number of changesets (and surfaces that count in the response). Is that the correct thing to sort by? (It could also make sense to sort by number of edits.)

giblet22 replied:
Is this what is being surfaced for users in the country pages on Scoreboard? I believe scoreboard has them sorted by number of edits. cc: @kamicut

sethfitz replied:
I'm not sure. @kamicut do you know where Scoreboard gets / uses that list?

sethfitz commented on sql/hashtag_statistics.sql:
```diff
@@ -1,159 +1,106 @@
+  ),
+  user_counts AS (
+    SELECT
+      -- TODO rank by edit count?
```
@giblet22 this produces the top 10 users by number of changesets (and surfaces that count in the response). Is that the correct thing to sort by? (It could also make sense to sort by number of edits.)

giblet22 replied:
Will confirm with customer, but I agree top hashtags by edits would make sense for a better picture overall.

sethfitz replied:
👍

sethfitz commented on sql/user_statistics.sql:
```diff
@@ -1,143 +1,152 @@
+    -- TODO top N?
+    countries,
+    -- TODO top N?
+    hashtags,
```
@giblet22 we really ought to cap the number of hashtags / countries / editors (which includes version numbers) when looking up user statistics. Should it be the top 10 of each (or more?)? Should they be sorted (to determine ranking) by number of changesets or edits?

giblet22 replied:
@mojodna Where are we currently surfacing these numbers, and if that number is capped, would it result in changes in Campaign metrics on Scoreboard and or user profiles?

sethfitz replied:
Just on user profiles in the list of "Top Hashtags"

sethfitz replied:
Countries are...in the list of countries (which is a constrained number, so maybe we leave that one and only cap hashtags).

mojodna commented:
@kamicut?

jpolchlo commented on sql/hashtag_statistics.sql:
```diff
@@ -1,159 +1,106 @@
+    FROM changesets
+    JOIN changesets_hashtags ON changesets.id = changesets_hashtags.changeset_id
+  ),
+  user_counts AS (
```
I'm not seeing where users 0 and 1 are being ignored here.  Did I miss it?

sethfitz replied:
https://github.com/azavea/osmesa-stat-server/pull/20/files#diff-ff794f69c23f9f516fe1dd05a5d23a70R6-R7

sethfitz committed:
Use edit counts for ranking country users and hashtags

This will now produce the top 10 users / hashtags by number of edits. In
addition, `user_edit_counts` (now `user_edits`) and `hashtag_edits` now
surface number of edits (not changesets). Changeset counts are surfaced
as `user_changesets` and `hashtag_changesets`.

sethfitz committed:
Use edit counts for ranking hashtag users

Also, surface both changeset and edit counts per-user.

sethfitz committed:
Use edit counts for ranking user countries, editors, and hashtags

Also splits out changesets and edits for each dimension.

sethfitz committed:
Merge branch 'develop' into jsonb-stats

mojodna commented:
@jpolchlo this is ready to go into `development` to support railways stats; can you take another look and merge if it looks good?

sethfitz committed:
Statistics by user and hashtag

sethfitz committed:
Refresh hashtag/user stats
